### DISCUSSION TITLE
I have several issues with Text node

### DISCUSSION THREAD

**Discussion by EviTRea** [olive-editor/olive#1942] [Ideas]
So I just used 0.2 nightly making a video, and this will probably be my future default editing software.
...Which is why I really hope the texts issues can be sorted out by the 0.2 release.
And, I'm not sure how opening an issue works, so I'll just post here.

To start with, I'll specify my use case: I use them to put subtitles onto my videos.
Yes there is already an dedicated "Subtitle track" exists, but text strips give me way more flexibility on how to display the texts so I need it.
Half of the issues here didn't exist in 0.1, they are newly introduced. 
I mean surely the something had to be risked to achieve viewport editing, but working things got broke is always sad.

**Issue 1: Which one am I editing?**
![Screenshot_20220528_121447](https://user-images.githubusercontent.com/32661913/170809438-04101f22-9575-4c19-894a-970d72ebeb3d.png)
Even If the text node isn't currently display, I can still edit them, like hide the track and move the timeline, it will still be edited.
Which is super confusing, especially when several strips are selected, I'd think the caching isn't catching up my text editing, only to realize I was editing another text strip the whole time.
![Screenshot_20220528_121738](https://user-images.githubusercontent.com/32661913/170809494-882e0896-ff26-43c6-9158-0652513c1b2f.png)
I mean, isn't the whole "Viewport editing update" point about instant previewing? 
Doesn't really make sense you can accidentally edit the texts without seeing it on screen.
**Potential solution**: Makes it only possible to edit on viewport ONLY when it's on viewport.

**Issue 2: Chinese typing**
I'm on KDE with fcitx5, and I can't type Chinese with my fcitx5-chewing input.
![Screenshot_20220528_121851](https://user-images.githubusercontent.com/32661913/170809519-18eba532-f3e8-4cbf-9457-045d6ad42e9e.png)
It feels like the interface just bypass the DE and captured my keyboard directly. Which is also the case on Steam, Spotify and even DaVinci Resolve.
Copy pasting from another text file works, but it's pretty annoying every time I want to add a character I'd have to switch to another application. (Which was part of the reason why I wanted to try outside Resolve.)
**Potential solution**: I have no idea how Qt works so I'll leave this one.

**Issue 3: Alignment and the box**
![Screenshot_20220528_121944](https://user-images.githubusercontent.com/32661913/170809549-85333a1c-a349-411d-a513-854aefb06c39.png)
THERE IS NO VERTIGO ALIGNMENT, NOOOOOOOOO
...Like, it's pretty common to have more than two lines of texts, and I happen to need them to align the single line came before it.
![Screenshot_20220528_122215](https://user-images.githubusercontent.com/32661913/170809624-52a2e59a-156b-4d39-b8e1-8d8defebd25f.png)
(Typo: TwoLine"s")
Having to visually or mathematically manually dragging them is one thing, and there aren't "Hold control to move it on one axis" or a clear way to precisely move them other than directly manipulate the text node position property.
**Potential Solution**: Just bring the 1.0 "Vertigo Alignment" option back?
![Screenshot_20220528_122324](https://user-images.githubusercontent.com/32661913/170809662-918c31b6-9d6f-4f20-8481-10698bd678cf.png)
Best case scenario is after setting up, I can just set an anchor in place, and it will be the dead center of whatever I write.
Also options under a "Text Box" category to toggle "Clip texts" and "Auto line switch" so the box size doesn't get in the way when people don't need it would be nice.

**Issue 4: Your format setting is no more**
So if I clear every character in the box, everything except horizontal alignment and line height will be set to default.
![Screenshot_20220528_123238](https://user-images.githubusercontent.com/32661913/170810036-4db55159-2597-451e-819a-f559205d90a2.png)
![Screenshot_20220528_124808](https://user-images.githubusercontent.com/32661913/170810416-8719e119-2c20-407a-9f28-fa993d7d8e2e.png)
Font is fine for me, since my system default font is also my subtitle font.
But the font size is default to 0.1. Like why? 
Like I said, I can't directly input Chinese, so I was just copying the whole sentence from somewhere else, Ctrl+A the text node and pasted it.
![Screenshot_20220528_122512](https://user-images.githubusercontent.com/32661913/170809701-4be8be4d-b2a4-4775-8fdc-b4be2d336e6e.png)
But doing so would result in the whole box becomes 0.1 in size, so I'd have to Ctrl+Z, paste it first, and remove the previous sentence after to keep the size and other settings.
![Screenshot_20220528_122624](https://user-images.githubusercontent.com/32661913/170809735-4a5a33dd-e3d5-47bb-af0b-51ecf79554a4.png)
Even if Chinese works, "clearing out existing texts" is a common enough action to be considered IMO.
**Potential solution**: Having format presets available, and only switch setting when user chose to load a preset.
(If auto switching makes sense in some context, make a "Load chosen default on empty" checkbox or something, I guess.)

And...That's all for now.
I still can't find a way to put gradient onto texts or the strokes, but if everything above is solved, I'll probably pass installing Resolve next time I distro hopping.
Even in current state, it's enough for my average editing process, although takes some workaround.
Super looking forward on this project.

**Comment by itsmattkc** [comment]
Thanks for testing this out, I think your issues are valid and worth resolving. I may create an issue from this so it can be put on the TODO list.
